src/main/java/io/cdap/plugin/ariba/source/AribaStructuredTransformer.java

```diff
@@ -0,0 +1,382 @@
+   *
+   * @param nonNullSchema non nullable Schema
+   * @param recBuilder    Structured record builder
+   * @param fieldName     Ariba objects's field name (may be encoded to remove CDF
+   *                      unsupported chars)
+   * @param fieldValTrim  trimmed value corresponding to an Ariba object's field
+   *                      name
+   */
+  private void processLogicalTypeVal(Schema nonNullSchema, StructuredRecord.Builder recBuilder, String fieldName,
+                                     String fieldValTrim) {
+
+    switch (Objects.requireNonNull(nonNullSchema.getLogicalType())) {
+      case DECIMAL:
+        fieldValTrim = handleMinusAtEnd(fieldValTrim);
+        recBuilder.setDecimal(fieldName, new BigDecimal(fieldValTrim).setScale(nonNullSchema.getScale()));
+        break;
+
+      case DATE:
+        recBuilder.setDate(fieldName, getSourceSpecificDateValue(fieldValTrim));
+        break;
+
+      case TIME_MICROS:
+        recBuilder.setTime(fieldName, getSourceSpecificTimeValue(fieldValTrim));
+        break;
+
+      case TIMESTAMP_MICROS:
+        ZonedDateTime zonedDateTime = null;
+        // Check if UTCLONG string having format yyyy-MM-dd HH:mm:ss'Z' does not
+        // start with default date value part 0000
+        if (!fieldValTrim.startsWith("0000")) {
+          String parsableTimestamp = fieldValTrim.replace(' ', 'T');
+          String nanoSecPart = "0";
+          zonedDateTime = ZonedDateTime.parse(parsableTimestamp, DateTimeFormatter.ISO_DATE_TIME);
+          zonedDateTime = zonedDateTime.plus(Long.parseLong(nanoSecPart), ChronoUnit.NANOS);
+        }
+        recBuilder.setTimestamp(fieldName, zonedDateTime);
+        break;
+
+      default:
+        recBuilder.set(fieldName, fieldValTrim);
+        break;
+    }
+  }
+
+  /**
+   * Process the value for field which is mapped to a {@code Schema.Type} and set
+   * into the {@code StructuredRecord.Builder}.
+   *
+   * @param fieldType   Schema field logical type
+   * @param recBuilder  Structured record builder
+   * @param fieldName   Ariba object's field name (may be encoded to remove CDF
+   *                    unsupported chars)
+   * @param fieldValStr value corresponding to an Ariba object's field
+   */
+  private void processTypeVal(Schema.Type fieldType, StructuredRecord.Builder recBuilder, String fieldName,
+                              String fieldValStr) {
+
+    String colValTrim = fieldValStr.trim();
+    switch (fieldType) {
+      case INT:
+        colValTrim = handleMinusAtEnd(colValTrim);
+        recBuilder.set(fieldName, Integer.parseInt(colValTrim));
+        break;
+
+      case LONG:
+        colValTrim = handleMinusAtEnd(colValTrim);
+        recBuilder.set(fieldName, Long.parseLong(colValTrim));
+        break;
+
+      case FLOAT:
+        colValTrim = handleMinusAtEnd(colValTrim);
+        // FLOAT values can contain exponential values.
+        recBuilder.set(fieldName, new BigDecimal(colValTrim));
```

Comment:
if the field can contain exponential values shouldn't the field type be DECIMAL?

Reply:
I have taken this reference from approved code of SAP Plugin but in our case we don't have FLOAT so removing this.